README.md

```diff
@@ -1 +1,235 @@
+
+#### Static assets must not contain anything that is environment-specific
+
+All of the leading application frameworks ([Angular CLI](https://github.com/angular/angular-cli/wiki/stories-application-environments), [Create React App](https://cli.vuejs.org/guide/mode-and-env.html#using-env-variables-in-client-side-code), [Ember CLI](https://ember-cli.com/user-guide/#Environments), [Vue CLI 3](https://cli.vuejs.org/guide/mode-and-env.html#using-env-variables-in-client-side-code)) recommend defining environment _values_ at _compile time_. This practice requires that the static assets are generated for each environment and regenerated for any change to an environment.
+
+_Immutable Web Applications_ reference environment _variables_ that are  defined on the global scope and that are set to values in the `index.html` that are unique to the deployment:
+
+```diff
+  import { Injectable } from '@angular/core';
+  import { HttpClient } from '@angular/common/http';
+
+- // remove environment constants that are defined in the codebase or at compile time
+- import { environment } from './environments/environment';
+
+  @Injectable()
+  export class DataService {
+    configUrl = ;
+
+    constructor(
+        private http: HttpClient
+        private
+        ) { }
+
+    getData() {
+-      return this.http.get(`${environment.API}/assets/data.json`);
++      // use globally scoped environment variables that are unique to the deployment
++      return this.http.get(`${window.env.API}/assets/data.json`);
+    }
+  }
+```
+
+#### Static assets must be hosted at locations that are unique and _independent of the web application environment(s)_.
+
+Static assets that do not contain anything environment-specific can be built once, published to once to a unique location, and then used in multiple environments of the web application.
+
+These static assets share the same qualities as hosted javascript libraries on content delivery networks (CDN) ([Google Hosted Libraries](https://developers.google.com/speed/libraries/), [cdnjs](https://cdnjs.com/), [jsDelivr](https://www.jsdelivr.com/), [UNPKG](https://unpkg.com/)):
+
+<code>https:/<span></span>/<span style="font-weight: bold">ajax.googleapis.c<span></span>om</span>/ajax/libs/jquery/<span style="font-weight: bold">3.3.1</span>/jquery.min.js</code>
+
+The location of the jquery library above, referenced by innumerable web applications, is both independent of the applications and uniquely versioned.
+
+<code>https:/<span></span>/<span style="font-weight: bold">assets.myapp.c<span></span>om</span>/apps/<span style="font-weight: bold">1.0.2</span>/main.js</code>
+
+Similarly, the location of the example the web application javascript file is unique and hosted at a location that is dedicated to static assets. The static asset web server is a repository for versions of the web application.
```

Comment:
This reads a bit weird...  Maybe this instead...

```suggestion
Similarly, the location of the web application javascript files are unique and hosted at a location that is dedicated to static assets. The static asset web server is a repository for versions of the web application.
```
Might also make sense to move the URL example below this paragraph.

Reply:
👍